app/components/npq_separation/admin/sub_navigation_component.rb

```diff
@@ -27,79 +29,70 @@ def sub_navigation_link(section)
         )
       end
 
-      def sub_navigation_item_link(node)
-        link_to(
-          node.name,
-          node.path,
-          class: "x-govuk-sub-navigation__link",
-          aria: { current: current?(node.prefix) },
+      def navigation_item_classes(section)
+        class_names(
+          "x-govuk-sub-navigation__section-item",
+          "x-govuk-sub-navigation__section-item--current" => current?(section.prefix),
         )
       end
 
-    private
-
-      def current?(prefix)
-        # return nil instead of false so Rails' link helper drops the
-        # attribute rather than setting "current='false'"
-        current_path.start_with?(prefix) || nil
-      end
-
-      def current_section
-        all_sections = [
-          Node.new(name: "Applications", prefix: "/npq-separation/admin/applications"),
-          Node.new(name: "Finance", prefix: "/npq-separation/admin/finance"),
-          Node.new(name: "Schools", prefix: "/npq-separation/admin/schools"),
-          Node.new(name: "Lead providers", prefix: "/npq-separation/admin/lead-providers"),
-          Node.new(name: "Settings", prefix: "/npq-separation/admin/settings"),
+      class SubNavigationStructure
```

Comment:
could we avoid having everything within the same file, and break every class into it's own file/spec as found slightly confusing to follow?

Reply:
Yeah, that's fair, updated.